src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -591,37 +425,58 @@ void TemporalMemory::compute(
                              activeColumnsUnsorted + activeColumnsSize);
   std::sort(activeColumns.begin(), activeColumns.end());
 
-  for (const ExcitedColumnData& excitedColumn : ExcitedColumns(activeColumns,
-                                                               activeSegments_,
-                                                               matchingSegments_,
-                                                               cellsPerColumn_))
+  const auto identity =
+    [](UInt c) { return c;};
```

Comment:
I went ahead and removed the silly `identity` lambda. Hopefully the other ones are less confusing, since they're actually using lambda-specific functionality (i.e. capturing locals via the `[&]` so that it can access `cellsPerColumn_`).

I think the codebase will get too messy if we put `// This is a lambda` above every lambda.